tests/system/test_dbapi.py

```diff
@@ -454,6 +456,49 @@ def test_read_timestamp_client_side_autocommit(self):
         read_timestamp_query_result_2 = self._cursor.fetchall()
         assert read_timestamp_query_result_1 != read_timestamp_query_result_2
 
+    @pytest.mark.parametrize("auto_commit", [False, True])
+    def test_batch_ddl(self, auto_commit, dbapi_database):
+        """Test batch ddl."""
+
+        self._conn._buffer_ddl_statements = False
+        if auto_commit:
+            self._conn.autocommit = True
+
+        self._cursor.execute("start batch ddl")
```

Comment:
I think it would make more sense if this statement already causes an error when `auto_commit=False`, instead of letting the error only happen when `run batch` is executed. Or is there some scenario where this could become a valid DDL batch?

Reply:
I think that my main takeaway from that is: It should not be possible to change the value of `auto_commit` while a batch is in progress. That should be the case for both for DDL batches and DML batches.

Take for example DML batches, and assume the following script:

```
set auto_commit=True;
start batch dml;
insert into foo (..);
set auto_commit=False;
insert into bar (...);
run batch;
rollback;
```

What should be the result of the above script?

Reply:
Made the change to not allow changing autocommit mode when a batch is in progress. Also raising error when starting a ddl batch when connection is not in autocommit mode